src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/WebJobs.Extensions.ServiceBus.csproj

```diff
@@ -31,17 +31,6 @@
     <PackageReference Include="WindowsAzure.Storage" Version="9.3.1" />
   </ItemGroup>
 
-  <ItemGroup>
-    <Compile Include="..\Microsoft.Azure.WebJobs.Host\Converters\AsyncConverter.cs" Link="Converters\AsyncConverter.cs" />
-    <Compile Include="..\Microsoft.Azure.WebJobs.Host\Converters\CompositeObjectToTypeConverter.cs" Link="Converters\CompositeObjectToTypeConverter.cs" />
-    <Compile Include="..\Microsoft.Azure.WebJobs.Host\Converters\ConversionResult.cs" Link="Converters\ConversionResult.cs" />
-    <Compile Include="..\Microsoft.Azure.WebJobs.Host\Converters\IAsyncObjectToTypeConverter.cs" Link="Converters\IAsyncObjectToTypeConverter.cs" />
-    <Compile Include="..\Microsoft.Azure.WebJobs.Host\Converters\IdentityConverter.cs" Link="Converters\IdentityConverter.cs" />
-    <Compile Include="..\Microsoft.Azure.WebJobs.Host\Converters\IObjectToTypeConverter.cs" Link="Converters\IObjectToTypeConverter.cs" />
-    <Compile Include="..\Microsoft.Azure.WebJobs.Host\Triggers\ITriggerDataArgumentBinding.cs" Link="Triggers\ITriggerDataArgumentBinding.cs" />
-    <Compile Include="..\Microsoft.Azure.WebJobs.Host\TypeUtility.cs" Link="TypeUtility.cs" />
-  </ItemGroup>
```

Comment:
Are you copying these into the ServiceBus repo? If so, I wonder if we want to create a separate nupkg for shared content files across the repos so we can keep them internal yet share the implementation...

https://medium.com/@attilah/source-code-only-nuget-packages-8f34a8fb4738.

I suspect we'll hit more-and-more of these. The ASP.NET team does this all over, sometimes for single files.

Reply:
If you're going to make a nupkg for them, they'll be public types, so why not just make them public in the core SDK?

Some of these are OK candidates for promoting to public types. Others (e.g. ITriggerDataArgumentBinding) are historical relics and shouldn't be made public.

Reply:
They're not public on nuget. It's a "source code only" package. In addition to the article I linked -- see here: https://twitter.com/davidfowl/status/1004237117608751104

It's useful for things like this -- Utility types or "legacy" types that you don't want to use to establish a public contract, but that you want to use internally across repos.

From the article:

> -The code you want to share is meant for setting up some internal things in your system and does not meant to be part of your publicly exposed types.
> -The code contains some useful helper, utility classes that’s used by other components within your system and/or can be useful for others as well, but only if consumed as source code, hence the type is internal.

Reply:
Ah, I see. Yes I agree then that that might be a good approach.